py/desispec/scripts/daily_processing.py

```diff
@@ -0,0 +1,239 @@
+                                                                                         flatjob, arcjob, lasttype,
+                                                                                         last_not_dither, internal_id,
+                                                                                         dry_run=dry_run, queue=queue)
+
+                prow = erow_to_prow(erow)
+
+                if camword is not None:
+                    prow['CAMWORD'] = camword
+                prow['INTID'] = internal_id
+                internal_id += 1
+                prow = define_and_assign_dependency(prow, arcjob, flatjob)
+                print(f"Processing: {prow}")
+                prow = create_and_submit(prow, dry_run=dry_run, queue=queue)
+                ptable.add_row(prow)
+
+                if curtype == 'flat' and flatjob is None and int(erow['SEQTOT']) < 5:
+                    flats.append(prow)
+                elif curtype == 'arc' and arcjob is None:
+                    arcs.append(prow)
+                elif curtype == 'science' and last_not_dither:
+                    sciences.append(prow)
+
+                lasttile = curtile
+                lasttype = curtype
+                last_not_dither = (prow['OBSDESC'] != 'dither')
+
+            time.sleep(10)
+            write_tables( [etable, ptable, unproc_table],
+                          fullpathnames=[exp_table_pathname,proc_table_pathname,unproc_table_pathname] )
+
+        time.sleep(300*speed_modifier)
+
+        if len(ptable) > 0:
+            ptable, nsubmits = update_and_recurvsively_submit(ptable,start_time=nersc_start,end_time=nersc_end,
+                                                              ptab_name=proc_table_pathname, dry_run=dry_run)
+
+            ## Exposure table doesn't change in the interim, so no need to re-write it to disk
+            write_table(ptable, tablename=proc_table_pathname)
+            time.sleep(300*speed_modifier)
+        
+    ## No more data coming in, so do bottleneck steps if any apply
+    ptable, arcjob, flatjob, internal_id = checkfor_and_submit_joint_job(ptable, arcs, flats, sciences, \
+                                                                         flatjob, arcjob, lasttype, last_not_dither,\
+                                                                         internal_id, dry_run=dry_run, queue=queue)
+
+
+    ## All jobs now submitted, update information from job queue and save
+    ptable = update_from_queue(ptable,start_time=nersc_start,end_time=nersc_end, dry_run=dry_run)
+    write_table(ptable, tablename=proc_table_pathname)
+
+    ## Now we resubmit failed jobs and their dependencies until all jobs have un-submittable end state
+    ## e.g. they either succeeded or failed with a code-related issue
+    ii,nsubmits = 0, 0
+    while ii < 4 and continue_looping(ptable['STATUS']):
```

Comment:
hardcoded 4 iterations?  What's the core stopping condition that it is trying to achieve here?

Reply:
As with many of the hardcoded numbers, it is a bit arbitrary, but with some thought put in. This code here is executed at the end of the night, looping over and resubmitting failed jobs. If a job has failed 4 times, odds are it will fail again on resubmission even if it seems like it should pass (based on exit codes).

Reply:
Still left with the thought-out but ultimately arbitrary number of iterations being 4. I t should be studied and optimized with test runs. For now this is commented out and punted to a future PR anyway, so I will resolve this here.